cw_evm/lib/evm_chain_wallet.dart

```diff
@@ -692,8 +693,29 @@ abstract class EVMChainWalletBase
   }
 
   @override
-  Future<String> signMessage(String message, {String? address}) async =>
-      bytesToHex(await _evmChainPrivateKey.signPersonalMessage(ascii.encode(message)));
+  Future<String> signMessage(String message, {String? address}) async {
+    final pk = await getPrivateKey(
+      mnemonic: _mnemonic,
+      privateKey: _hexPrivateKey,
+      password: _password,
+    );
+    return EthSigUtil.signPersonalMessage(
+      privateKey: bytesToHex(pk.privateKey),
```

Comment:
better use `_evmChainPrivateKey` it's already set, and it handles hardware wallet case

Reply:
made the change for the sign function but I'm unsure of what the flow really looks like for hardware wallets @konstantinullrich do any additional changes need to be made to the UI? will a prompt just appear if a hardware wallet is connected and signMessage is called or is there something else that needs to be added?